contracts/PlasmaCore.sol

```diff
@@ -72,13 +73,23 @@ library PlasmaCore {
                 txindex: input[1].toUint(),
                 oindex: input[2].toUint()
             });
+            
+            // check for empty inputs - disallow gaps
+            if (decodedTx.inputs[i].blknum == 0) emptySeen[0] = true;
```

Comment:
We are not consistent on definition of what constitutes gap.

https://github.com/omisego/elixir-omg/blob/84c407396f6da0b57514cc4871c41efe7dfc8cf0/apps/omg/lib/state/transaction.ex#L230

Reply:
To summarize internal discussion: input {0, 0, 0} is incorrect as any other from block 0, however it's allowed because it's how we represent empty (not provided) value.
And analogically we apply the same for outputs: all zeros means empty.